redisgraph/execution_plan.py

```diff
@@ -0,0 +1,61 @@
+class Operation:
+    def __init__(self, name, args=None):
+        self.name = name
+        self.args = args
+        self.children = []
+
+    def append_child(self, child):
+        self.children.append(child)
+        return self
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Operation):
+            return False
+
+        if self.name != o.name or self.args != o.args or len(self.children) != len(o.children):
+            return False
+
+        for i in range(len(self.children)):
+            if not self.children[i] == o.children[i]:
+                return False
+
+        return True
+
+
+class ExecutionPlan:
+    def __init__(self, plan):
+        self.plan = plan
+        self.structured_plan = self._operation_tree()
+
+    def __str__(self) -> str:
+        return "\n".join(self.plan)
+
+    def _operation_tree(self):
```

Comment:
wouldn't you prefer converting this function into a recursive function instead of maintaining a stack ?

Reply:
no because then I need to know each call how much rows it worked on
this way is simpler